src/modules/Sidebar/Sidebar.js

```diff
@@ -63,20 +69,45 @@ class Sidebar extends Component {
   static Pushable = SidebarPushable
   static Pusher = SidebarPusher
 
-  startAnimating = (duration = 500) => {
-    clearTimeout(this.stopAnimatingTimer)
+  componentDidMount() {
+    if (this.state.visible) eventStack.sub('click', this.handleDocumentClick)
+  }
 
-    this.setState({ animating: true })
+  componentWillUnmount() {
+    if (this.state.visible) eventStack.unsub('click', this.handleDocumentClick)
+  }
 
-    this.stopAnimatingTimer = setTimeout(() => this.setState({ animating: false }), duration)
+  componentDidUpdate(prevProps, { visible: prev }) {
+    const { visible: next } = this.state
+
+    if (prev === next) return
+
+    this.startAnimating()
+    if (next) eventStack.sub('click', this.handleDocumentClick)
+    if (!next) eventStack.unsub('click', this.handleDocumentClick)
   }
 
-  componentWillReceiveProps(nextProps) {
-    if (nextProps.visible !== this.props.visible) {
-      this.startAnimating()
-    }
+  startAnimating = (duration = 500) => {
+    this.setState({ animating: true }, () => {
+      clearTimeout(this.stopAnimatingTimer)
+      this.stopAnimatingTimer = setTimeout(this.handleAnimationEnd, duration)
+    })
   }
 
+  handleAnimationEnd = () => {
+    if (!this.state.visible) _.invoke(this.props, 'onHidden')
+    this.setState({ animating: false })
+  }
+
+  handleDocumentClick = (e) => {
+    if (!this.state.visible || this.ref.contains(e.target)) return
```

Comment:
In the event that the result of the click removes the target node from the DOM, it will fail to be detected here.  The ref will not contain the `e.target` because the target node was removed before this is invoked.

We wrote `doesNodeContainClick` for this specific usage.  You can import it from `lib` and use it like so:

```jsx
doesNodeContainClick(this.ref, e)
```

You can search the codebase for more usage examples, or look at the tests in `doesNodeContainClick-test.js`.